s3/pkg/datastore/aws/aws.go

```diff
@@ -60,9 +65,60 @@ func (myc *s3Cred) IsExpired() bool {
 	return false
 }
 
+func (ad *AwsAdapter) BucketCreate(ctx context.Context, in *pb.Bucket) error {
+
+	log.Info("Bucket create is called in aws service")
+
+	Credentials := credentials.NewStaticCredentials(ad.backend.Access, ad.backend.Security, "")
+	configuration := &aws.Config{
+		Region:      aws.String(DefaultRegion),
+		Endpoint:    aws.String(ad.backend.Endpoint),
+		Credentials: Credentials,
+	}
+	svc := awss3.New(session.New(configuration))
+
+	input := &awss3.CreateBucketInput{
+		Bucket: aws.String(in.Name),
+		CreateBucketConfiguration: &awss3.CreateBucketConfiguration{
+			LocationConstraint: aws.String(ad.backend.Region),
+		},
+		ACL: aws.String(in.Acl.CannedAcl),
+	}
+
+	_, err := svc.CreateBucketWithContext(ctx, input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			case awss3.ErrCodeBucketAlreadyExists:
+				log.Error(awss3.ErrCodeBucketAlreadyExists, aerr.Error())
+			case awss3.ErrCodeBucketAlreadyOwnedByYou:
+				log.Error(awss3.ErrCodeBucketAlreadyOwnedByYou, aerr.Error())
+			default:
+				log.Error(aerr.Error())
+			}
+		} else {
+			// Print the error, cast err to awserr.Error to get the Code and
+			// Message from an error.
+			log.Error(err.Error())
+		}
+		return err
+	}
+
+	// Wait until bucket is created before finishing
+	err = svc.WaitUntilBucketExists(&awss3.HeadBucketInput{
```

Comment:
Since we are making it sync, the API call will wait for the Bucket to be created successfully. What if the API call take time more than expected? Wait?